packages/backend/src/peripherals/database/LivenessRepository.test.ts

```diff
@@ -151,4 +151,62 @@ describe(LivenessRepository.name, () => {
       expect(result).toEqual([records[0], records[2]])
     })
   })
+
+  describe(
+    LivenessRepository.prototype.getWithTypeDistinctTimestamp.name,
+    () => {
+      it('join and returns data with type', async () => {
+        const result = await repository.getWithTypeDistinctTimestamp(
+          LIVENESS_CONFIGS[0].projectId,
+        )
+        const expected = [
+          {
+            timestamp: DATA[0].timestamp,
+            projectId: LIVENESS_CONFIGS[0].projectId,
+            type: LIVENESS_CONFIGS[0].type,
+          },
+        ]
+
+        expect(result).toEqual(expected)
+      })
+
+      it('filters out transactions with the same timestamp', async () => {
+        const NEW_DATA = [
```

Comment:
this test will be more readable if you delete all records in the beginning and add all them here